File: backend/app/schemas/line.py
from datetime import datetime
from typing import Optional
from uuid import UUID

from pydantic import BaseModel, Field


# ── LineMessage ──

class LineMessageOut(BaseModel):
    id: UUID
    line_user_id: str
    lead_id: Optional[UUID] = None
    message_text: str
    direction: str
    message_type: str
    source_type: str
    reply_status: Optional[str] = None
    created_at: datetime

    model_config = {"from_attributes": True}


# ── Conversation ──

class LineConversationOut(BaseModel):
    line_user_id: str
    lead_id: Optional[UUID] = None
    lead_name: Optional[str] = None
    messages: list[LineMessageOut]
    latest_message_at: Optional[datetime] = None


# ── Reply ──

class LineReplyRequest(BaseModel):
    reply_token: str
    message_text: str = Field(..., max_length=5000)
    line_user_id: str | None = None


class LinePushRequest(BaseModel):
    line_user_id: str
    message_text: str = Field(..., max_length=5000)


# ── AI Reply Suggestion ──

class LineAIReplyResponse(BaseModel):
    suggested_reply: str
    lead_context: Optional[dict] = None

File: frontend/src/types/line.ts
export interface LineMessageOut {
  id: string;
  line_user_id: string;
  lead_id: string | null;
  message_text: string;
  direction: 'incoming' | 'outgoing';
  message_type: string;
  source_type: string;
  reply_status: string | null;
  created_at: string;
}

export interface LineConversationOut {
  line_user_id: string;
  lead_id: string | null;
  lead_name: string | null;
  messages: LineMessageOut[];
  latest_message_at: string | null;
}

export interface LineConversationSummary {
  line_user_id: string;
  lead_id: string | null;
  lead_name: string;
  lead_status: string;
  latest_message_at: string;
  message_count: number;
}

export interface LineAIReplyResponse {
  suggested_reply: string;
  lead_context: Record<string, unknown> | null;
}

export interface LineReplyRequest {
  reply_token: string;
  message_text: string;
}

export interface LinePushRequest {
  line_user_id: string;
  message_text: string;
}
